runtime/moonbeam/src/lib.rs

```diff
@@ -831,7 +831,14 @@ impl_runtime_apis! {
 			source: TransactionSource,
 			tx: <Block as BlockT>::Extrinsic,
 		) -> TransactionValidity {
-			Executive::validate_transaction(source, tx)
+			match &tx.function {
+				Call::Balances(_) => InvalidTransaction::Call.into(),
+				Call::CrowdloanRewards(_) => InvalidTransaction::Call.into(),
+				Call::Democracy(_) => InvalidTransaction::Call.into(),
+				Call::Ethereum(_) => InvalidTransaction::Call.into(),
+				Call::EVM(_) => InvalidTransaction::Call.into(),
```

Comment:
Is it possible to use the `BaseCallFilter` itself here? So there is no risk of them falling out of sync. Maybe something like

```rust
if <Runtime as frame_system::Config>::BaseCallFilter::filter_or_whatever_the_method_is_called(&tx) {
  Executive::validate_transaction(source, tx)
} else {
  Call::EVM(_) => InvalidTransaction::Call.into()
}
```

Reply:
It's cool. I'm thinking `BaseCallFilter` is redundant if we have this check because any transactions included in a block must first be in the transaction pool and this change prevents those transactions from entering the pool.

Is there any way for a malicious node to not follow the filtering logic in this PR and compose blocks that do not require `Executive::validate_transaction` returns `ValidTransaction` for all transactions? I don't think so, but want to make sure.

Reply:
Afaik this will only be there on early stages where we assume only known nodes will be authoring blocks. That being said, I'd love some feedback on the whole approach to know if there is something we are missing @crystalin maybe link the PR to Shawn?